hazelcast-hibernate53/src/main/java/com/hazelcast/hibernate/local/LocalRegionCache.java

```diff
@@ -0,0 +1,362 @@
+        maybeNotifyTopic(key, null, null);
+    }
+
+    protected Object createMessage(final Object key, final Object value, final Object currentVersion) {
+        return new Invalidation(key, currentVersion);
+    }
+
+    @SuppressWarnings("Duplicates")
+    protected void maybeInvalidate(final Object messageObject) {
+        final Invalidation invalidation = (Invalidation) messageObject;
+        final Object key = invalidation.getKey();
+        if (key == null) {
+            // Invalidate the entire region cache.
+            cache.clear();
+        } else if (versionComparator == null) {
+            // For an unversioned entity or collection we can only invalidate the entry.
+            cache.remove(key);
+        } else {
+            // For versioned entities we can avoid the invalidation if both we and the remote node know the version,
+            // AND our version is definitely equal or higher.  Otherwise, we have to just invalidate our entry.
+            final Expirable value = cache.get(key);
+            if (value != null) {
+                maybeInvalidateVersionedEntity(key, value, invalidation.getVersion());
+            }
+        }
+    }
+
+    @SuppressWarnings("Duplicates")
+    void cleanup() {
+        final int maxSize;
+        final long timeToLive;
+        if (config != null) {
+            maxSize = config.getMaxSizeConfig().getSize();
+            timeToLive = config.getTimeToLiveSeconds() * SEC_TO_MS;
+        } else {
+            maxSize = MAX_SIZE;
+            timeToLive = CacheEnvironment.getDefaultCacheTimeoutInMillis();
+        }
+
+        final boolean limitSize = maxSize > 0 && maxSize != Integer.MAX_VALUE;
+        if (limitSize || timeToLive > 0) {
+            final List<EvictionEntry> entries = searchEvictableEntries(timeToLive, limitSize);
+            final int diff = cache.size() - maxSize;
+            final int evictionRate = calculateEvictionRate(diff, maxSize);
+            if (evictionRate > 0 && entries != null) {
+                evictEntries(entries, evictionRate);
+            }
+        }
+    }
+
+    void maybeNotifyTopic(final Object key, final Object value, final Object version) {
+        if (topic != null) {
+            topic.publish(createMessage(key, value, version));
+        }
+    }
+
+    private int calculateEvictionRate(final int diff, final int maxSize) {
+        return diff >= 0 ? (diff + (int) (maxSize * BASE_EVICTION_RATE)) : 0;
+    }
+
+    private MessageListener<Object> createMessageListener() {
+        return new MessageListener<Object>() {
+
+            @Override
+            public void onMessage(final Message<Object> message) {
+                maybeInvalidate(message.getMessageObject());
+            }
+        };
+    }
+
+    private void evictEntries(final List<EvictionEntry> entries, final int evictionRate) {
+        // Only sort the entries if we're going to evict some
+        Collections.sort(entries);
```

Comment:
I think this can be `entries.sort()` in Java 8

Reply:
Done.